dns/push.sh

```diff
@@ -29,6 +29,19 @@ ALL_ZONES=(
     "${PROD_ZONES[@]}"
 )
 
+DRY_RUN=false
+
+# Read the arguments to determine if should only perform a dry run
+while getopts "d" o; do
+  case "${o}" in
+    d)
+      DRY_RUN=true
+      ;;
+    *)
+      ;;
+  esac
+done
+
```

Comment:
I'm not familiar with this kind of arguments parsing but can we do `d | dry-run )` instead of `d`?

That's my way of parsing arguments as a reference (with some more flags/arguments which we'll have to add in the future (ref. #783):

```bash
function parse_args() {
  # positional args
  args=()

  # named args
  while [ "$#" -gt 0 ]; do
      case "$1" in
          -d | --dry-run )          DRY_RUN=true;           ;;
          -f | --from-container )   FROM_CONTAINER=true;    ;;
          -c | --config-file )      CONFIG_FILE="$2";       shift;;
          * )                       args+=("$1")            # if no match, add it to the positional args
      esac
      shift # move to next kv pair
  done

  # restore positional args
  set -- "${args[@]}"

  # set defaults
  if [[ -z "${CONFIG_FILE:-}" ]]; then
      CONFIG_FILE="octodns-config.yaml";
  fi
}

parse_args "$@";
```

Reply:
It looks good but I would liek to ask you for one more thing. As we are currently not using arguments `--from-container` and `--config-file` could you remove it for now? It would look like:

```bash
DRY_RUN=false

function parse_args() {
  # positional args
  args=()

  # named args
  while [ "$#" -gt 0 ]; do
      case "$1" in
          -d | --dry-run )          DRY_RUN=true;           ;;
          * )                       args+=("$1")            # if no match, add it to the positional args
      esac
      shift # move to next kv pair
  done

  # restore positional args
  set -- "${args[@]}"
}

parse_args "$@";
```

Following up #783 (step no. 2 in particulary) we'll add this functionality and I would love to work with you on that topic too if you want. :-)

Reply:
I have removed the two unused flags.
I'd be happy to help with adding the that functionality